procedures: Setting Authorization callback URL in GitHub OAuth app to Che OAuth callback URL (#2177)
codenvy/che-docs: modules/administration-guide/partials/proc_configuring-github-oauth.adoc

.Procedure

> * Create a link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] using {prod-short} URL as the value for the application `Homepage URL`. The default values are `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/`

Reviewer: ```suggestion
* Create a link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] using {prod-short} URL as the value for the application `Homepage URL`. The default values are `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/`. When entering the `Authorization callback URL`:
```

> * Create a link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] using {prod-short} URL as the value for the application `Homepage URL`. The default values are `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/`. When entering the `Authorization callback URL`:

Reviewer: ```suggestion
. Create an link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] with the following URLs:

* Enter the {prod-short} URL as `Homepage URL`. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/`.

* Enter the `Authorization callback URL` as follows:
```

> ** If the {devworkspace} engine is not enabled the value for `Authorization callback URL` should be {identity-provider} GitHub endpoint URL. The default value is `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/github/endpoint`.

Reviewer: ```suggestion
** If the {devworkspace} engine is not enabled, set the value for `Authorization callback URL` to the {identity-provider} GitHub endpoint URL. The default value is `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/github/endpoint`.
```

> ** If the {devworkspace} engine is not enabled, set the value for `Authorization callback URL` to the {identity-provider} GitHub endpoint URL. The default value is `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/github/endpoint`.

Reviewer: ```suggestion
** If the {devworkspace} engine is not enabled, enter the {identity-provider} GitHub endpoint URL in the `Authorization callback URL` field. The default value is `++https://++keycloak-{prod-namespace}.__<domain>__/auth/realms/{prod-deployment}/broker/github/endpoint`.
```

> ** If the {devworkspace} engine is enabled  the value for `Authorization callback URL` should be {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`.

Reviewer: ```suggestion
** If the {devworkspace} engine is enabled, set the value for `Authorization callback URL` to the {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`.
```

> ** If the {devworkspace} engine is enabled, set the value for `Authorization callback URL` to the {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`.

Reviewer: ```suggestion
** If the {devworkspace} engine is enabled, enter the {prod-short} OAuth callback URL in the `Authorization callback URL` field. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback`.
```

.Procedure

. Create an link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] with the following URLs:

* Enter the {prod-short} URL as `Homepage URL`. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/`.

* Enter the `Authorization callback URL` as follows:

> ** If the {devworkspace} engine is not enabled, enter the {identity-provider} GitHub endpoint URL in the `Authorization callback URL` field. The default value is `++https://++keycloak-{prod-namespace}.__<domain>__/auth/realms/{prod-deployment}/broker/github/endpoint`.

Reviewer: ```suggestion
** If the {devworkspace} engine is not enabled, enter the {identity-provider} GitHub endpoint URL as `Authorization callback URL`. The default value is `++https://++keycloak-{prod-namespace}.__<domain>__/auth/realms/{prod-deployment}/broker/github/endpoint`.
```

> ** If the {devworkspace} engine is enabled, enter the {prod-short} OAuth callback URL in the `Authorization callback URL` field. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback`.

Reviewer: ```suggestion
** If the {devworkspace} engine is enabled, enter the {prod-short} OAuth callback URL as `Authorization callback URL`. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback`.
```

* Create a link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] using {prod-short} URL as the value for the application `Homepage URL` and {identity-provider} GitHub endpoint URL as the value for Authorization callback URL. The default values are `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/` and `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/github/endpoint` respectively, where `__<DOMAIN>__` is {orch-name} cluster domain.

> * When {prod-short} is deployed using native authentication, the value for `Authorization callback URL` should be {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`.

Reviewer: What is "native authentication"?

Reviewer: ```suggestion
* When {prod-short} is deployed using native authentication, set the `Authorization callback URL` value to {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`.
```

Reviewer: "_When {prod-short} is deployed using native authentication_" - is "_when_" or "_if_" more appropriate here? "_When_" is what happens always; "_if_" is what may happen under certain conditions.

Reviewer 2: this sentence is an alternative to the sentence above. If the {devworkspace} engine is enabled we should put one callback our, if not  - another.

Reviewer: In such case, would it be better to split the second half of the existing paragraph into two alternatives? For example:

_Create a link:https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app[OAuth application in GitHub] using {prod-short} URL as the value for the application `Homepage URL` and {identity-provider} GitHub endpoint URL as the value for Authorization callback URL:_
* [edit the beginning of the last sentence in the existing paragraph, for example:] _If the {devworkspace} engine is not enabled... The default values are `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/` and `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/github/endpoint` respectively, where `__<DOMAIN>__` is {orch-name} cluster domain._
* [edit the beginning of the new paragraph, for example:] _If the {devworkspace} engine is not enabled... When {prod-short} is deployed using native authentication, the value for `Authorization callback URL` should be {prod-short} OAuth callback URL. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<DOMAIN>__/api/oauth/callback`._

Reviewer 2: Sounds great. Let me check if I can change the pr.

Reviewer 3: It's not clear to me, what is the current proposal. If we're going to deprecate che-server engine in next release, we should not make much effort to describe everything for all cases. Also I think it's enough to say something like "For devworkspace engine", no need to say the native authentication stuff, because it's now the default with devworkspaces.

Reviewer: The docs still have content related to che-server engine, so we need clarity about which content needs to be removed and which content is going to stay.
